libraries/AP_Camera/AP_RunCam.cpp

```diff
@@ -0,0 +1,774 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  implementation of RunCam camera protocols 
+
+  With thanks to betaflight for a great reference
+  implementation. Several of the functions below are based on
+  betaflight equivalent functions
+
+  RunCam protocol specification can be found at https://support.runcam.com/hc/en-us/articles/360014537794-RunCam-Device-Protocol
+ */
+#include "AP_RunCam.h"
+
+#include <AP_Math/AP_Math.h>
+#include <AP_HAL/AP_HAL.h>
+#include <GCS_MAVLink/GCS.h>
+#include <AP_Logger/AP_Logger.h>
+#include <AP_OSD/AP_OSD.h>
+
+const AP_Param::GroupInfo AP_RunCam::var_info[] = {
```

Comment:
Maybe add an enable param for this, so that people who aren't using runcams don't have the parameters coming down?

Reply:
I considered this, but nothing is enabled unless you special the runcam protocol on a serial device. If I add enable you have to both enable and select the protocol to get stuff working - which seems a little onerous, but happy to do this if that's the way we configure things. For comparison blheli protcol support does not have an enable flag.